src/utils/viewer/viewer.cpp

```diff
@@ -664,13 +682,24 @@ void Viewer::drawEvent() {
   Mn::GL::defaultFramebuffer.clear(Mn::GL::FramebufferClear::Color |
                                    Mn::GL::FramebufferClear::Depth);
 
-  // step physics at a fixed rate
+  // Move and look actions should occur at a rate of 60 per second,
+  //  which is the optimal frame rate (so 1 event per frame in the optimal case)
+  //  In the non-optimal case, the number of actions per frame is given by repetitionsToPerform:
   timeSinceLastSimulation += timeline_.previousFrameDuration();
-  if (timeSinceLastSimulation >= 1.0 / 60.0 &&
-      (simulating_ || simulateSingleStep_)) {
-    simulator_->stepWorld(1.0 / 60.0);
-    timeSinceLastSimulation = 0.0;
-    simulateSingleStep_ = false;
+  int repetitionsToPerform = timeSinceLastSimulation * 60.0;
```

Comment:
Let's rename this "60" magic number: `constexpr float agentActionsPerSecond = 60.0f`. And lets rename `repetitionsToPerform` to `numAgentActions`.